src/app/data/AdData.jsx

```diff
@@ -0,0 +1,229 @@
+import { containsEmail, extractEmail, formatDate, isValidEmail, mailtoInString } from "@/app/_common/utils/utils";
```

Comment:
Den här filen borde väl hellre ligga under `stilling`-foldern? Kanske `stilling/_data`?

Reply:
Har ikke bekreftet men har en følelse av at kanskje den kommer til å eventuelt bli brukt andre steder? FavouritesButton f.eks. blir brukt nå både i <Search> og i <Ad> og idag forventer en `stilling` object som kanskje kunne vært `adData` i fremtiden. Jeg ville ikke koblet denne bare til `stilling` enda.

Reply:
Att lägga den under `/stilling` betyder ju inte att Favoruites inte kan bruke den. Det är mest för att gruppera relaterad logik i närheten av varandra. Saker är ofta avhengig av varandra uansett, så det blir aldrig "helt clean" utan att det är dependencies mellan moduler.

Är egentligen [Low Coupling, High Cohesion](https://medium.com/clarityhub/low-coupling-high-cohesion-3610e35ac4a6) jag är ute efter :)

Reply:
Ok ja, er forholdsvis enig men sjekket litt grundigere og akkurat nå så erstatter vi `fetchAd` med `getAdData` for `stilling` komponenten. Et kjapp søk for `fetchAd` så ser vi at den brukes flere steder som `trekk-soknad` og `rapporter-annonse`. Med mindre vi har tenkt å ha egne data objekter for dem så tenker jeg at denne burde ligge under `_common` da det virker som den "kan" bli tatt i bruk flere steder.

Reply:
Nästan allt har avhengigheter till något annat på något sätt. Gör man så, så blir det fort att all logik ligger i `_common` och det blir rörigt och oöversiktligt.  Att den blir brukt på fler ställen tycker jag inte är ett argument för att lägga den i `_common`. Att den heter `AdData` däremot, tycker jag är ett starkt argument för att den ska grupperas med andra Ad-ting.

`_common` tänker jag mer är generella saker, som kan användas överallt och som inte helt passar in i domänen. Hjälp med datumformatering, t.ex. Ad har ju en naturlig plats på ett annat sätt.

Om det hade varit första gången du jobbade i applikationen. Någon ber dig leta efter data-layern för stillinger (och du kan inte söka efter den). Hade du då manuellt navigerat dig in i `_common`-foldern eller `stilling`-foldern? Jag vet var jag hade provat först iaf :)

Reply:
Jeg også foretrekker at det er by feature. `/app/_common` synes jeg ikke følger noen pattern, der er mest som en roteskuff.

Man kunne argumentert for at vi kunne hatt `/app/_data`, men er ikke sikker på at data-laget gjelder hele applikasjonen på lang sikt. 

Derfor stemmer jeg for å ha det i `/app/stilling/_data`

Reply:
Jeg flytter dem til /app/stilling/_data da. Er ikke uenig med ideen at data skal være nærme der den brukes men min følelse er fortsatt at adData er mer en "felles" objekt enn en "stilling" objekt, hvertfall slik vi har brukt fetchAd/elasticSearch data idag.